packages/drivers/odsp-driver/src/exposeJoinSessionInfo.ts

```diff
@@ -0,0 +1,28 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { IDocumentServiceFactory, IResolvedUrl } from "@fluidframework/driver-definitions";
+import { ISocketStorageDiscovery } from "./contractsPublic";
+import { getJoinSessionCacheKey, getOdspResolvedUrl } from "./odspUtils";
+import { OdspDocumentServiceFactoryCore } from "./odspDocumentServiceFactoryCore";
+
+/**
+ * Api which returns the current join session response stored in non persistent cache, if present
+ * @param factory - odsp driver factory
+ * @param resolvedUrl - resolved url for container
+ * @returns - Current join session response stored in cache. Undefined if not present.
+ */
+export async function getJoinSessionInfo(
+	factory: IDocumentServiceFactory,
+	resolvedUrl: IResolvedUrl,
+): Promise<ISocketStorageDiscovery | undefined> {
+	assert(factory instanceof OdspDocumentServiceFactoryCore, "factory type is not recognized");
```

Comment:
Would throwing a `UsageError` be better here? The factory being something that the consumer passes to a FF API, it feels like a more ergonomic API if we tell the consumer what they did wrong. Based on the scenario we have right now, the component might not be able to do anything to fix the issue at that point, but in a scenario where the host/loader was the one actually calling this instead of making it available to the component, a `UsageError` definitely feels more correct IMO.

Also slightly concerned about the usual `instanceof` caveats. Not sure if the factory instance that the host has could be passed across iframe/window contexts by the time it gets to the component who ends up triggering this call. @vladsud do you know?

Reply:
Removed instanceof and replaced the type with OdspDriverServiceFactory.